packages/api/tests/endpoints/test-granules-updateGranulesAndFilesCollection.js

```diff
@@ -0,0 +1,339 @@
+  recursivelyDeleteS3Bucket,
+  s3PutObject,
+} = require('@cumulus/aws-client/S3');
+const { createSnsTopic } = require('@cumulus/aws-client/SNS');
+const indexer = require('@cumulus/es-client/indexer');
+const {
+  createTestIndex,
+  cleanupTestIndex,
+} = require('@cumulus/es-client/testUtils');
+const { randomString, randomId } = require('@cumulus/common/test-utils');
+const { constructCollectionId } = require('@cumulus/message/Collections');
+const models = require('../../models');
+
+const { request } = require('../helpers/request');
+
+// Dynamo mock data factories
+const {
+  createFakeJwtAuthToken,
+  setAuthorizedOAuthUsers,
+} = require('../../lib/testUtils');
+
+const testDbName = `granules_${cryptoRandomString({ length: 10 })}`;
+
+let accessTokenModel;
+let jwtAuthToken;
+
+process.env.AccessTokensTable = randomId('token');
+process.env.stackName = randomId('stackname');
+process.env.system_bucket = randomId('system-bucket');
+process.env.TOKEN_SECRET = randomId('secret');
+process.env.backgroundQueueUrl = randomId('backgroundQueueUrl');
+
+// import the express app after setting the env variables
+const { app } = require('../../app');
+
+/**
+ * Simulate granule records post-collection-move for database updates test
+ *
+ * @param {Knex | Knex.Transaction} knexOrTransaction - DB client or transaction
+ * @param {Array<Object>} [granules] - granule records to update
+ * @param {Object} [collection] - current collection of granules used for translation
+ * @param {string} [collectionId] - collectionId of current granules
+ * @param {string} [collectionId2] - collectionId of collection that the granule is moving to
+ * @returns {Array<Object>} - list of updated apiGranules (moved to new collection)
+ */
+const simulateGranuleUpdate = async (knex, granules, collection, collectionId, collectionId2) => {
+  const movedGranules = [];
+  for (const granule of granules) {
+    const postMoveApiGranule = await translatePostgresGranuleResultToApiGranule(knex, {
+      ...granule,
+      collectionName: collection.name,
+      collectionVersion: collection.version,
+    });
+    postMoveApiGranule.collectionId = collectionId2;
+    postMoveApiGranule.updatedAt = Date.now();
+    postMoveApiGranule.lastUpdateDateTime = new Date().toISOString();
+    for (const apiFile of postMoveApiGranule.files) {
+      apiFile.bucket = apiFile.bucket.replace(collectionId, collectionId2);
+      apiFile.key = apiFile.key.replace(collectionId, collectionId2);
+      apiFile.updatedAt = Date.now();
+    }
+    movedGranules.push(postMoveApiGranule);
+  }
+  return movedGranules;
+};
+
+test.before(async (t) => {
+  const { knexAdmin, knex } = await generateLocalTestDb(
+    testDbName,
+    migrationDir
+  );
+  process.env.CMR_ENVIRONMENT = 'SIT';
+  process.env = {
+    ...process.env,
+    ...localStackConnectionEnv,
+    PG_DATABASE: testDbName,
+  };
+
+  const topicName = randomString();
+  const { TopicArn } = await createSnsTopic(topicName);
```

Comment:
It looks like we're creating an SNS topic here, are we using this?     Consider removing

Reply:
Oh, I see, we have an SNS update buried in the granule update logic, and this avoids the need to stub or do something more convoluted.    Got it, thanks for the comment :thumbsup: